astarte_device.c

```diff
@@ -862,9 +862,25 @@ static void send_introspection(astarte_device_handle_t device)
 
     esp_mqtt_client_handle_t mqtt = device->mqtt_client;
 
-    char introspection_string[512] = { 0 };
-    size_t total_written = 0;
     struct astarte_list_head_t *item;
+    size_t introspection_size = 0;
+    LIST_FOR_EACH(item, &device->interfaces_list)
+    {
+        struct astarte_ptr_list_entry_t *entry
+            = GET_LIST_ENTRY(item, struct astarte_ptr_list_entry_t, head);
+        const astarte_interface_t *interface = entry->value;
+        char interface_entry[128];
+        snprintf(interface_entry, 128, "%s:%d:%d;", interface->name, interface->major_version,
```

Comment:
what about: "`strlen(interface_entry) + major_digits + minor_digits + 2;`"?

Reply:
+3 (two `:` and an ending `;`)